src/custom/hooks/useWrapCallback.ts

```diff
@@ -0,0 +1,112 @@
+import { TransactionResponse } from '@ethersproject/providers'
+import { getChainCurrencySymbols } from '@src/custom/utils/xdai/hack'
+import { Currency, CurrencyAmount, currencyEquals, ETHER, WETH } from '@uniswap/sdk'
+import { Contract } from 'ethers'
+import { useMemo } from 'react'
+import { tryParseAmount } from 'state/swap/hooks'
+import { useTransactionAdder } from 'state/transactions/hooks'
+import { useCurrencyBalance } from 'state/wallet/hooks'
+import { useActiveWeb3React } from 'hooks/index'
+import { useWETHContract } from 'hooks/useContract'
+
+export enum WrapType {
+  NOT_APPLICABLE,
+  WRAP,
+  UNWRAP
+}
+
+interface WrapUnwrapCallback {
+  wrapType: WrapType
+  execute?: undefined | (() => Promise<void>)
+  inputError?: string
+}
+
+type TransactionAdder = ReturnType<typeof useTransactionAdder>
+
+interface GetWrapUnwrapCallback {
+  isWrap: boolean
+  balance?: CurrencyAmount
+  inputAmount?: CurrencyAmount
+  addTransaction: TransactionAdder
+  wethContract: Contract
+}
+
+const NOT_APPLICABLE = { wrapType: WrapType.NOT_APPLICABLE }
+
+function _getWrapUnwrapCallback(params: GetWrapUnwrapCallback): WrapUnwrapCallback {
+  const { native, wrapped } = getChainCurrencySymbols()
+  const { isWrap, balance, inputAmount, addTransaction, wethContract } = params
+  const symbol = isWrap ? native : wrapped
+
+  // Check if user has enough balance for wrap/unwrap
+  const sufficientBalance = (inputAmount && balance && !balance.lessThan(inputAmount)) || false
+
+  // Create wrap/unwrap callback if sufficient balance
+  let wrapUnwrapCallback: (() => Promise<void>) | undefined
+  if (sufficientBalance && inputAmount) {
+    let wrapUnwrap: () => TransactionResponse
+    let summary: string, symbol: string
+
+    if (isWrap) {
+      wrapUnwrap = () => wethContract.deposit({ value: `0x${inputAmount.raw.toString(16)}` })
+      summary = `Wrap ${inputAmount.toSignificant(6)} ${native} to ${wrapped}`
+    } else {
+      wrapUnwrap = () => wethContract.withdraw(`0x${inputAmount.raw.toString(16)}`)
+      summary = `Unwrap ${inputAmount.toSignificant(6)} ${wrapped} to ${native}`
+    }
+
+    wrapUnwrapCallback = async () => {
+      try {
+        const txReceipt = await wrapUnwrap()
+        addTransaction(txReceipt, { summary })
+      } catch (error) {
+        const actionName = WrapType.WRAP ? 'wrapping' : 'unwrapping'
+        console.error(`Error ${actionName} ${symbol}`, error)
```

Comment:
Where is `symbol` value set?

Reply:
up, https://github.com/gnosis/gp-swap-ui/pull/120/files#diff-c0dc186d4d0cface14e71753887eeb6b89f0bf0f2db6dc1f2c406531e11b152cR39